channeldb/graph_test.go

```diff
@@ -2132,40 +2132,85 @@ func TestChannelEdgePruningUpdateIndexDeletion(t *testing.T) {
 
 	// Now that both edges have been updated, if we manually check the
 	// update index, we should have an entry for both edges.
+	timestamps := map[uint64]struct{}{
+		uint64(edge1.LastUpdate.Unix()): {},
+		uint64(edge2.LastUpdate.Unix()): {},
+	}
 	if err := db.View(func(tx *bolt.Tx) error {
 		edges := tx.Bucket(edgeBucket)
 		if edges == nil {
 			return ErrGraphNoEdgesFound
 		}
-		edgeIndex := edges.Bucket(edgeIndexBucket)
-		if edgeIndex == nil {
-			return ErrGraphNoEdgesFound
-		}
 		edgeUpdateIndex := edges.Bucket(edgeUpdateIndexBucket)
 		if edgeUpdateIndex == nil {
 			return ErrGraphNoEdgesFound
 		}
 
-		var edgeKey [8 + 8]byte
+		numEntries := edgeUpdateIndex.Stats().KeyN
+		if numEntries != 2 {
+			return fmt.Errorf("expected %v entries in the update "+
+				"index, got %v", 2, numEntries)
+		}
+
+		return edgeUpdateIndex.ForEach(func(k, _ []byte) error {
+			t := byteOrder.Uint64(k[:8])
+			if _, ok := timestamps[t]; !ok {
+				return fmt.Errorf("found unexpected timestamp "+
+					"%d", t)
+			}
 
-		byteOrder.PutUint64(edgeKey[:8], uint64(edge1.LastUpdate.Unix()))
-		byteOrder.PutUint64(edgeKey[8:], edge1.ChannelID)
+			return nil
+		})
+	}); err != nil {
+		t.Fatal(err)
+	}
 
-		if edgeUpdateIndex.Get(edgeKey[:]) == nil {
-			return fmt.Errorf("first edge not found in update " +
-				"index")
+	// Now, we'll update the edge policies to ensure the old timestamps are
+	// removed from the update index.
+	edge1.Flags = 2
+	edge1.LastUpdate = time.Now()
```

Comment:
time.Now() could be a source of instable test runs. Even though it isn't very exciting in this scenario, you could consider using a fixed value, as I did in 938c2f7

Reply:
We currently don't any fixed values when creating random node vertices and edge policies within the series of `channeldb` tests. Should those be extended as well?

Reply:
I'd say so, but maybe not at this moment. To me it is more something to first agree on that we all want it and then slowly move in that direction. For my own unit tests, I use deterministic node keys:

https://github.com/lightningnetwork/lnd/blob/16b5a67c3aa99907a4dff8d9b67175466557ffc0/routing/pathfind_test.go#L364